agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -101,6 +102,13 @@ void monitor_thread::stop_monitor_thread()
             mon_hal_int_events = nullptr;
         }
 
+        if (mon_hal_nl_events) {
+            remove_socket(mon_hal_nl_events);
+            delete mon_hal_nl_events;
```

Comment:
By looking at the rest of the file I wonder why we have to resort to explicitly calling delete here, though that's not related to this PR.

Reply:
@kantera800
I'm not sure myself if this is needed

Reply:
from looking at the code remove_socket just isn't strong enough